src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5356,6 +5364,21 @@ private bool DetermineIfDefinitelyTypeArgumentList(bool isDefinitelyTypeArgument
             return isDefinitelyTypeArgumentList;
         }
 
+        private bool IsPossibleTypeInTypeArgumentList()
+        {
+            if (!this.IsPossibleType())
+            {
+                return false;
+            }
+
+            // If type argument list is part of return type then also check whether this "possible type" is a member's name
+            if ((_termState & TerminatorState.IsEndOfReturnType) == TerminatorState.IsEndOfReturnType && this.IsEndOfReturnType()) {
```

Comment:
this doesn't seem right to me. _termState isn't really meant to be directly queried.  @gafter can probably tell you the right pattern to use here.

Reply:
The correct pattern to use is to implement the check inside `IsTerminator()` and simply call that.

---
In reply to: [170162496](https://github.com/dotnet/roslyn/pull/24911#discussion_r170162496) [](ancestors = 170162496)

Reply:
I think this whole method could be
``` c#
    return this.IsPossibleType() && !IsTerminator();
```
But I'm not sure that the other kinds of terminators sought in IsTerminator() will not break things; that would need to be tested.

---
In reply to: [171018381](https://github.com/dotnet/roslyn/pull/24911#discussion_r171018381) [](ancestors = 171018381,170162496)

Reply:
Please try this suggestion. I'm not happy with this line as it stands, and I'd prefer you try the simpler form.

---
In reply to: [171022774](https://github.com/dotnet/roslyn/pull/24911#discussion_r171022774) [](ancestors = 171022774,171018381,170162496)